cmd/deploy_gateway_name.go

```diff
@@ -0,0 +1,73 @@
+// Package cmd for parsing command line arguments
+package cmd
+
+import (
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/cobra"
+	command "github.com/threefoldtech/grid3-go/internal/cmd"
+	"github.com/threefoldtech/grid3-go/workloads"
+	"github.com/threefoldtech/zos/pkg/gridtypes/zos"
+)
+
+// deployGatewayNameCmd represents the deploy gateway-name command
+var deployGatewayNameCmd = &cobra.Command{
+	Use:   "gateway-name",
+	Short: "Deploy a gateway name proxy",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		name, err := cmd.Flags().GetString("name")
```

Comment:
we can make a function for fqdn and name inputs together except fqdn field

also we can have a gateway command that has subcommands `name` and `fqdn` and we can put the shared flags on gateway cmd